setup.sh

```diff
@@ -1,19 +1,50 @@
 #!/bin/bash
+set -x
 
+# This script installs juju version 2 and all the dependencies.
+
+# Refresh the potentially stale apt cache.
 apt-get update -qq
+# Install software tools such as add-apt-repository.
 apt-get install -qy software-properties-common
+# Install the ppa for Juju and charm tools.
 apt-add-repository -y ppa:juju/stable
+# Refresh the apt-cache with the new Juju PPA.
 apt-get update -qq
 
-## don't install lxc & local till we get local support
-#apt-get install -qy lxc iptables juju-local
-
-## actually, we need juju-local since the local provider
-## can be used by bootstrapping from outside the container
-apt-get -qy install juju-local
-apt-get -qy install juju-core juju-quickstart juju-deployer python-jujuclient
-apt-get -qy install tmux charm-tools openssh-client
-apt-get -qy install virtualenvwrapper python-dev cython git
+# Install juju-local since the local provider can be used by bootstrapping 
+# from outside the container.
+apt-get install -qy \
+    byobu \
+    cython \
+    git \
+    juju \
+    openssh-client \
+    python-dev \
+    sudo \
+    vim \
+    virtualenvwrapper
 
-useradd -m ubuntu
+# Add ubuntu to the passwordless sudo file.
 echo "ubuntu ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/juju-users
+
+# Get the version of Juju installed.
+JUJU_VERSION=`juju version`
+# The home directory for the ubuntu user.
+HOME=/home/ubuntu
+# Set some environment variables used by Juju.
+RC=${HOME}/.bashrc
+cat << EOF > $RC
+export PROJECT_HOME=${HOME}
+# The directory to look for charms.
+export JUJU_REPOSITORY=${HOME}
```

Comment:
if my earlier comment about mounting JUJU_REPO over $HOME results in a change to the JUJU_REPO path inside the docker container, be sure to update this setup.sh with that location too.